src/traits.rs

```diff
@@ -121,6 +121,18 @@ pub trait RootContext: Context {
     fn on_queue_ready(&mut self, _queue_id: u32) {}
 
     fn on_log(&mut self) {}
+
+    fn create_http_context(&self, _context_id: u32) -> Box<dyn HttpContext> {
+        Box::new(EmptyHttpContext)
+    }
+
+    fn create_stream_context(&self, _context_id: u32) -> Box<dyn StreamContext> {
+        Box::new(EmptyStreamContext)
+    }
+
+    fn get_type(&self) -> ContextType {
+        ContextType::RootContext
+    }
```

Comment:
Shouldn't those functions be split into a separate trait?

Reply:
to avoid the need for a `RootContext`? I'm not sure about that. I think `RootContext` (by its name) already has a notion of controlling its child contexts, so this is what I would expect, instead of another meta-struct that creates all contexts.

Also, the main reason to implement this was so that the `on_configure` calls could be passed on to child contexts easily by the `RootContext`. if we move context creation out of the root context again, it'll again lose the ability to map `RootContext` to e.g. `HttpContext` - except if the user explicitly keeps a mapping in that struct